### PR TITLE
Return parse errors to clients and increase message size limits

### PR DESCRIPTION
**Pull request by agis**

### CODE REVIEW

File: server.go [hunk 1]
```diff
@@ -51,6 +51,11 @@ type Server struct {
 }
 
 func NewServer(timeout time.Duration) *Server {
+	// The default is 65KB, however Kafka messages might typically be
+	// much larger than this, so we the limit.
```
Comment (konstantinoskostis) [inline server.go]: `much larger than this, so we the limit` -> `... so we increase the limit`

File: server.go [hunk 2]
```diff
@@ -51,6 +51,14 @@ type Server struct {
 }
 
 func NewServer(timeout time.Duration) *Server {
+	// The default is 65KB, however Kafka messages might typically be
+	// much larger than this, so we bump the limit to 512MB.
+	//
+	// Keep in mind that the Redis protocol also specifies strings may be
+	// up to 512MB, although there are plans to make them even bigger
+	// (see https://github.com/antirez/redis/issues/757).
+	redisproto.MaxBulkSize = 1024 * 512
```
Comment (apoikos) [inline server.go]: This is 512 **k**B I guess :)

File: server.go [hunk 3]
```diff
@@ -313,11 +318,10 @@ func (s *Server) Handle(ctx context.Context, conn net.Conn) {
 				writeErr = writer.WriteError("Command not supported")
 			}
 		}
-		if command.IsLast() {
+		if parseErr != nil || command.IsLast() {
 			writer.Flush()
 		}
-		if writeErr != nil {
-			// TODO(agis) log these errors
+		if parseErr != nil || writeErr != nil {
```
Comment (0xdiba) [inline server.go]: Should we at least log `writeErr`s ?  I think the previous TODO was valid :)

File: server.go [hunk 4]
```diff
@@ -313,11 +318,10 @@ func (s *Server) Handle(ctx context.Context, conn net.Conn) {
 				writeErr = writer.WriteError("Command not supported")
 			}
 		}
-		if command.IsLast() {
+		if parseErr != nil || command.IsLast() {
 			writer.Flush()
```
Comment (0xdiba) [inline server.go]: If we decide on logging `writeErr`s, let's catch this errors too.

File: server.go (hunk 2 above)
Comment (0xdiba) [reply server.go]: Should we have Redis as a direction of max message size or think of the messages we pass along to Kafka so we do not propagate errors? Following Kafka's `message.max.bytes` seems more reasonable.

Either way, we should probably document our "protocol" max size in the README.

File: server.go (hunk 3 above)
Comment (agis) [reply server.go]: They are returned to the clients, so no need to also log them in the server.

File: server.go (hunk 2 above)
Comment (agis) [reply server.go]: Bumped to 32MB.

File: server.go (hunk 3 above)
Comment (agis) [reply server.go]: Added logging.

File: server.go [hunk 5]
```diff
@@ -51,6 +51,14 @@ type Server struct {
 }
 
 func NewServer(timeout time.Duration) *Server {
+	// The default is 65KB, however Kafka messages might typically be
```
Comment (0xdiba) [inline server.go]: nit:  `The default is 64KB`

**Pull request merged**